File: examples/key/decrypt_aes/decrypt_aes.java
import com.bloock.sdk.client.EncryptionClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.AesDecrypter;
import com.bloock.sdk.entity.AesEncrypter;
import com.bloock.sdk.entity.Record;
import java.nio.charset.StandardCharsets;

public class App {
  public static void main(String[] args) throws Exception {
    String payload = "This will be encrypted";

    // encryption ...

    System.out.println("Trying to decrypt with the valid password");
    RecordClient recordClient = new RecordClient();
    KeyClient keyClient = new KeyClient();

    LocalKey localKey = keyClient.newLocalKey(KeyType.Aes256);

    // To decrypt a record during the building process
    // we build a record from the encrypted record and add a decrypter
    Record decryptedRecord = recordClient.fromRecord(encryptedRecord)
        .withDecrypter(new AesDecrypter(localKey))
        .build();

    // To decrypt an already encrypted record independently
    EncryptionClient encryptionClient = new EncryptionClient();
    decryptedRecord = encryptionClient.decrypt(encryptedRecord, new AesDecrypter(localKey));

    System.out.println(
        "Decrypted payload: " + new String(decryptedRecord.retrieve(), StandardCharsets.UTF_8));

  }
}

File: examples/key/decrypt_aes/decrypt_aes.go
package main

import (
	"fmt"
	"log"

	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/encryption"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
	"github.com/bloock/bloock-sdk-go/v2/entity/record"
)

func main() {
	recordClient := client.NewRecordClient()
	encryptionClient := client.NewEncryptionClient()
	keyClient := client.NewKeyClient()

	key, err := keyClient.NewLocalKey(key.Aes256)
	if err != nil {
		log.Println(err)
	}

	// encryption ...
	var encryptedRecord record.Record

	fmt.Println("Trying to decrypt with the valid password")

	// To decrypt a record during the building process
	// we build a record from the encrypted record and add a decrypter
	decryptedRecord, err := recordClient.FromRecord(encryptedRecord).
		WithDecrypter(encryption.NewEncrypterWithLocalKey(key)).
		Build()
	if err != nil {
		log.Println(err)
	}

	// To decrypt an already encrypted record independently
	decryptedRecord, err = encryptionClient.Decrypt(encryptedRecord, encryption.NewEncrypterWithLocalKey(key))
	if err != nil {
		log.Println(err)
	}

	fmt.Println("Decryption successful")
	fmt.Println("Decrypted payload: " + string(decryptedRecord.Retrieve()))
}
